src/tools/msvc.jam

```diff
@@ -907,7 +925,7 @@ local rule configure-really ( version ? : options * )
             # version from the path.
             # FIXME: We currently detect both Microsoft Visual Studio 9.0 and
             # 9.0express as 9.0 here.
-            if [ MATCH "(Microsoft Visual Studio 15)" : $(command) ]
+            if [ MATCH "(14.10)" : $(command) ]
```

Comment:
It's logic to deduce the VS version from a given path. `14.10` is the only stable path part (`MSVC\14.10` at most).
There are other patterns in this block RE: [line 965](https://github.com/refack/build/blob/4973c19e47efbe338664145818fdc00fb24759e9/src/tools/msvc.jam#L965)
VS2017 setup has been reworked extensively...